### ISSUE TITLE
no need to have sigmoid function if to use cross_entropy

### ISSUE THREAD

**Issue by yfeizhang** [opendataval/opendataval#25]
### Versions

In [logistic_regression.py](https://github.com/opendataval/opendataval/blob/main/opendataval/model/logistic_regression.py), have been added sigmoid function at the end (x = F.sigmoid(x)), then use crossentropy function as criterion (criterion = F.binary_cross_entropy if self.num_classes == 2 else F.cross_entropy). In pytorch, crossentropy function encompasses the computation of sigmoid function, this will affect the performance at my end. Therefore, sigmoid function (x = F.sigmoid(x)) should be removed.

### Operating System

Linux

**Comment by kevinfjiang** [comment]
Hi Yifei, just trying to understand your problem a bit better. it looks like we can change from `F.sigmoid` to `F.softmax` as softmax is equivalent to sigmoid for classes of size 2. I don't think cross entropy does handle the sigmoid function, however, so I don't think it makes sense to remove the function entirely, but to include a check

**Comment by kevinfjiang** [comment]
Small fix added let me know what you think!

**Comment by yfeizhang** [comment]
Hi Kevin, there is a trick playing in cross entropy loss function of pytorch, which has implemented logsoftmax before computing the loss. You can check documentation from pytorch again to make sure. Therefore, adding another layer is equivalent to do softmax twice. It should be removed. You can also refer to other standard repos which use cross entropy loss function of pytorch and generally not have the last layer of softmax because cross entropy loss function has covered it.
________________________________
